test/jdk/com/sun/jndi/ldap/LdapSSLHandshakeFailureTest.java

```diff
@@ -93,110 +142,152 @@ public static void main(String args[]) throws Exception {
                     .port(server.getPortNumber())
                     .buildUnchecked().toString());
 
-            if (hasCustomSocketFactory) {
-                env.put("java.naming.ldap.factory.socket", args[0]);
-                env.put("com.sun.jndi.ldap.connect.timeout", "1000");
+            if (customSocketFactory != null) {
+                env.put("java.naming.ldap.factory.socket", customSocketFactory);
             }
 
-            if (args.length == 2 && !hasCustomSocketFactory) {
-                env.put("com.sun.jndi.ldap.connect.timeout", args[0]);
+            if (connectTimeout != null) {
+                env.put("com.sun.jndi.ldap.connect.timeout", connectTimeout.toString());
             }
-
             env.put(Context.SECURITY_PROTOCOL, "ssl");
             env.put(Context.SECURITY_AUTHENTICATION, "Simple");
             env.put(Context.SECURITY_PRINCIPAL, "cn=principal");
             env.put(Context.SECURITY_CREDENTIALS, "123456");
             LdapContext ctx = null;
             try {
                 ctx = new InitialLdapContext(env, null);
-            } catch (Exception e) {
-                if (CustomSocketFactory.customSocket.closeMethodCalledCount() > 0
-                        && hasCustomSocketFactory
-                        && Boolean.valueOf(args[1])) {
-                    System.out.println(SOCKET_CLOSED_MSG);
+            } catch (NamingException e) {
+                if (customSocketFactory != null) {
+                    System.out.println("Caught expected Exception with custom SocketFactory (no SSL Socket).");
+                    if (CustomSocketFactory.customSocket.closeMethodCalledCount() <= 0) {
+                        System.err.println("Custom Socket was not closed.");
+                        System.exit(-1);
```

Comment:
Can we update test not to use `System.exit` and replace it with throwing new `RuntimeException`, something like:
```suggestion
                        throw new RuntimeException("Custom Socket was not closed");
```

Reply:
Done.